src/nmea.cpp

```diff
@@ -0,0 +1,1041 @@
+		7       numSV     -       numeric         10 Number of satellites used (range: 0-99)
+		8         HDOP    -       numeric         0.83 Horizontal Dilution of Precision
+		9         alt     m       numeric         111.1 Altitude above mean sea level
+		10        sep    m        numeric         45.6 Geoid separation: difference between ellipsoid and mean sea level UBX-18010854 - R05 Advance Information Page 18 of 262 u-blox ZED-F9P Interface Description - Manual GNS continued
+		11    diffAge    s        numeric         - Age of differential corrections (blank when DGPS is not used)
+		12 diffStation   -        numeric         - ID of station providing differential corrections (blank when DGPS is not used)
+		13 navStatus    -         character         V Navigational status indicator (V = Equipment is not providing navigational status information) NMEA v4.10 and above only
+		14 cs - hexadecimal *71   Checksum
+		15 <CR><LF> - character - Carriage return and line feed
+		*/
+		double utc_time = 0.0;
+		double lat = 0.0, lon = 0.0;
+		char pos_Mode[5] = {'N', 'N', 'N', 'N', 'N'};
+		int num_of_sv = 0;
+		float alt = 0.f;
+		float hdop = 0.f;
+		char ns = '?', ew = '?';
+		int i = 0;
+		NMEA_UNUSED(pos_Mode);
+
+		if (bufptr && *(++bufptr) != ',') { utc_time = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { lat = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { ns = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { lon = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { ew = *(bufptr++);}
+
+		do {
+			pos_Mode[i] =  *(bufptr);
+			i++;
+
+		} while (*(++bufptr) != ',');
+
+		if (bufptr && *(++bufptr) != ',') { num_of_sv = strtol(bufptr, &endp, 10); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { hdop = strtof(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { alt = strtof(bufptr, &endp); bufptr = endp; }
+
+		if (ns == 'S') {
+			lat = -lat;
+		}
+
+		if (ew == 'W') {
+			lon = -lon;
+		}
+
+		/* convert from degrees, minutes and seconds to degrees */
+		_gps_position->lat = static_cast<int>((int(lat * 0.01) + (lat * 0.01 - int(lat * 0.01)) * 100.0 / 60.0) * 10000000);
+		_gps_position->lon = static_cast<int>((int(lon * 0.01) + (lon * 0.01 - int(lon * 0.01)) * 100.0 / 60.0) * 10000000);
+		_gps_position->hdop = hdop;
+		_gps_position->alt = static_cast<int>(alt * 1000);
+		_sat_num_gns = static_cast<int>(num_of_sv);
+
+		if (!_POS_received && (_last_POS_timeUTC < utc_time)) {
+			_last_POS_timeUTC = utc_time;
+			_POS_received = true;
+		}
+		_ALT_received = true;
+		_SVNUM_received = true;
+
+		// mavlink_log_info(&mavlink_log_pub, "GNS> posMode %s/sv_num %d/ hdop %.2f",
+		// (char*)(pos_Mode),
+		// (int)(sat_num_gns),
+		// (double)(hdop));
+
+	}	else if ((memcmp(_rx_buffer + 3, "RMC,", 4) == 0) && (uiCalcComma >= 11)) {
```

Comment:
Did you run the style check from PX4 Autopilot?
```suggestion
	} else if ((memcmp(_rx_buffer + 3, "RMC,", 4) == 0) && (uiCalcComma >= 11)) {
```